src/vector/ops/set.rs

```diff
@@ -4,20 +4,27 @@ use crate::vector::Geometry;
 ///
 /// These methods provide set operations over two geometries, producing a new geometry.
 impl Geometry {
-    /// Compute intersection.
+    /// Computes the [geometric intersection][intersection] of `self` and `other`.
     ///
-    /// Generates a new geometry which is the region of intersection of
-    /// the two geometries operated on. Call intersects (Not yet implemented)
-    /// to check if there is a region of intersection.
-    /// Geometry validity is not checked. In case you are unsure of the
-    /// validity of the input geometries, call IsValid() before,
-    /// otherwise the result might be wrong.
+    /// Generates a new geometry which is the region of intersection of the two geometries operated on.
+    ///
+    /// # Notes
+    /// * If you only need to determine if two geometries intersect and don't require
+    /// the resultant region, use [`Geometry::intersects`].
+    /// * Geometry validity is not checked, and invalid geometry will generate unpredictable results.
+    /// Use [`Geometry::is_valid`] if validity might be in question.
+    /// * If GEOS is *not* enabled, this function will always return `None`.
+    /// You  may check for GEOS support with [`VersionInfo::has_geos`][has_geos].
     ///
     /// # Returns
-    /// `Some(geometry)` if both Geometries contain pointers
-    /// `None` if either geometry is missing the GDAL pointer, or there is an error.
+    /// * `Some(geometry)`: a new `Geometry` representing the computed intersection
+    /// * `None`: when the geometries do not intersect or result could not be computed
```

Comment:
Along with [this](https://github.com/georust/gdal/pull/394/files#r1186926701), frustrated to see the overloading of `Option` to handle both the empty geometry case as well as the error case. Not sure if we should try to fix this on the Rust side.

Reply:
> when the geometries do not intersect

At least GEOS (didn't test with SFCGAL) returns an empty geometry (not `NULL`) here.